File: design-patterns/go/chain_of_responsibility/interview/technical_interview.go
// technical_interview.go

package main

import "fmt"

type TechnicalInterview struct {
	nextInterview Interview
}

func NewTechnicalInterview(nextInterview Interview) (technicalInterview *TechnicalInterview) {
	technicalInterview = &TechnicalInterview{}
	technicalInterview.nextInterview = nextInterview
	return
}

func (technicalInterview *TechnicalInterview) Execute() {
	// Ask all questions for technical interview
	// Perform any other action required for technical interview
	fmt.Println("Ask technical interview questions")

	// Execute the next interview set while creating new struct
	if technicalInterview.nextInterview != nil {
		technicalInterview.nextInterview.Execute()
	}
}

File: design-patterns/ts/chain-of-responsibility/interview/technical-interview.ts
// technical-interview.ts

import Interview from "./interview";

class TechnicalInterview extends Interview {
    constructor(nextInterview: Interview | null) {
        super(nextInterview);
    }

    execute(): void {
        // Ask all questions for technical interview
        // Perform any other action required for technical interview
        console.log("Ask technical interview questions")

        // Execute the next interview set while creating new struct
        if (this.nextInterview) {
            this.nextInterview.execute();
        }
    }

}

export default TechnicalInterview;